Feature/homework 3

VtulkinVladimir opened the pull request:
Приложение - визитка

VtulkinVladimir committed:
разработал приложение согласно заданию

реализованны три экрана, навигация через табБар

VtulkinVladimir committed:
прочие файлы

c0dedbear commented on MyCV/MyCV/Hobbys/Hobbys.swift:
```diff
@@ -0,0 +1,16 @@
+//
+import UIKit
+
+class MyHobbys
```
Почему класс, а не структура?

VtulkinVladimir replied:
честно говоря как-то не задумывался даже, делал как делал до этого. ну наверное чтобы не копировался каждый раз когда используется. Хотя в моем случае он используется один раз, и меняться там ничего не запланировано тогда смысла от класса получается и нет

c0dedbear replied:
Всегда когда возможно лучше использовать value type.

c0dedbear reviewed (changes requested):
Фотка огонь)

Ошибки - те про которые писал в телеге, final'ы, магические цифры.
Так же есть баг в альбомной ориентации на первом экране - часть текста не видно.
Ну и tabBar без иконок совсем уж печально выглядит.

c0dedbear commented on MyCV/MyCV/Hobbys/HobbysViewController.swift:
```diff
@@ -0,0 +1,16 @@
+{
+	override func viewDidLoad() {
+		super.viewDidLoad()
+		self.view = HobbysView()
```
viewDidLoad не то место где нужно менять корневую вью. Это нужно делать в loadVIew.

c0dedbear commented on MyCV/MyCV/MainTabBar.swift:
```diff
@@ -0,0 +1,30 @@
+		let item = UITabBarItem(title: "Info", image: nil, tag: 1)
+		infoVc.tabBarItem = item
+		
+		let skillsVc = SkillsViewController(nibName: nil, bundle: nil)
```
Нет смысла задавать такие иниты . Можно просто InfoViewController()  и т.д.

VtulkinVladimir commented:
не могу найти как ответить на коментарий насчет бага по первому экрану. дело в том, что там вообще не планировалась альбомная ориентация, поэтому я даже и не проверял как смотрится при ней. видимо забыл её отключить. или надо сделать чтоб и в альбомной работало?

VtulkinVladimir committed:
Хранение информации, модификаторы, смена метода

переделал хранение информации для экранов в структуры, со статическими свойствами.
Добавил к классам, которые не наследую final.
переписал замещение корневой вьюхи в другой метод.

VtulkinVladimir committed:
исправление "магических чисел"

добавил энумы, описывающие хранимые свойства

VtulkinVladimir committed:
"развеселил" tabBar

добавил иконки на tabBar. использовал другой инициализатор для контроллеров

VtulkinVladimir committed:
автоматические файлы

c0dedbear reviewed (approved):
Принято
